### PR TITLE
[eagleplatform] extract mp4 url and fix thumbnail url

### PR DESCRIPTION
**Pull request by remitamine**

### CODE REVIEW

**Comment by dstftw** [comment]
Now response of the first `_download_json` call is not handled by `_handle_error`.

**Comment by remitamine** [comment]
returned the old code to handle all _download_json requests.

**Comment by dstftw** [comment]
```
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'http://lentaru.media.eagleplatform.com/index/player?player=new&record_id=227304&player_template_id=5201', u'-v']
[debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
[debug] youtube-dl version 2015.09.22
[debug] Git HEAD: 88e1161
[debug] Python version 2.6.6 - Windows-2003Server-5.2.3790-SP2
[debug] exe versions: ffmpeg N-74386-g542562e, ffprobe N-74386-g542562e, rtmpdump 2.4
[debug] Proxy map: {}
[EaglePlatform] 227304: Downloading JSON metadata
[EaglePlatform] 227304: Downloading m3u8 JSON
ERROR: Unable to download JSON metadata: HTTP Error 404: Not Found (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

  File "C:\Dev\git\youtube-dl\master\youtube_dl\extractor\common.py", line 327, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "C:\Dev\git\youtube-dl\master\youtube_dl\YoutubeDL.py", line 1865, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "C:\python\python266\lib\urllib2.py", line 397, in open
    response = meth(req, response)
  File "C:\python\python266\lib\urllib2.py", line 510, in http_response
    'http', request, response, code, msg, hdrs)
  File "C:\python\python266\lib\urllib2.py", line 435, in error
    return self._call_chain(*args)
  File "C:\python\python266\lib\urllib2.py", line 369, in _call_chain
    result = func(*args)
  File "C:\python\python266\lib\urllib2.py", line 518, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
```

**Comment by remitamine** [comment]
i don't know why you get 404 error.
it works for me.
```
python2 __main__.py -v 'http://lentaru.media.eagleplatform.com/index/player?player=new&record_id=227304&player_template_id=5201'
[debug] System config: []
[debug] User config: [u'--external-downloader', u'aria2c', u'--external-downloader-args', u'-x 16', u'-f', u'[height<=?720]', u'--sub-lang', u'ar,arME,en', u'--write-sub', u'--no-check-certificate']
[debug] Command-line args: [u'-v', u'http://lentaru.media.eagleplatform.com/index/player?player=new&record_id=227304&player_template_id=5201']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.09.22
[debug] Git HEAD: 1ecd4f1
[debug] Python version 2.7.10 - Linux-4.1.6-1-ARCH-x86_64-with-glibc2.2.5
[debug] exe versions: ffmpeg 2.8, ffprobe 2.8, rtmpdump 2.4
[debug] Proxy map: {}
[EaglePlatform] 227304: Downloading JSON metadata
[EaglePlatform] 227304: Downloading m3u8 JSON
[EaglePlatform] 227304: Downloading m3u8 information
[EaglePlatform] 227304: Downloading mp4 JSON
[debug] Invoking downloader on u'http://b39.servers.eaglecdn.com/lentaru/2015-03-06/54f99992800be_mobile_360p.mp4?st=D9_mOMjsC1dHoBQ1aRDk1g&e=1443296171'
[download] Destination: Навальный вышел на свободу-227304.mp4
[debug] aria2c command line: aria2c -c -x 16 --out 'Навальный вышел на свободу-227304.mp4.part' --header 'Accept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7' --header 'Accept-Language: en-us,en;q=0.5' --header 'Accept-Encoding: gzip, deflate' --header 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' --header 'User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20150101 Firefox/20.0 (Chrome)' --check-certificate=false -- 'http://b39.servers.eaglecdn.com/lentaru/2015-03-06/54f99992800be_mobile_360p.mp4?st=D9_mOMjsC1dHoBQ1aRDk1g&e=1443296171'
[#b829e8 64KiB/4.7MiB(1%) CN:5 DL:32KiB ETA:2m27s]
```

**Comment by dstftw** [comment]
Because `scheme` in `self._proto_relative_url` must be explicitly set to `http:` that you've removed.

**Comment by remitamine** [comment]
i removed it because when i try to run the tests without the scheme parameter they run without a problem so i thought that it's not important

File: youtube_dl/extractor/eagleplatform.py [hunk 1]
```diff
@@ -78,13 +81,18 @@ def _real_extract(self, url):
...
+        secure_m3u8 = self._proto_relative_url(media['sources']['secure_m3u8']['auto'], 'http:')
 
+        m3u8_url = self._get_video_url(secure_m3u8, video_id, 'Downloading m3u8 JSON')
         formats = self._extract_m3u8_formats(
-            m3u8_data['data'][0], video_id,
+            m3u8_url, video_id,
             'mp4', entry_protocol='m3u8_native')
+
+        mp4_url = self._get_video_url(
+            secure_m3u8.replace("m3u8", "mp4").replace("hlsvod", "mp4").replace("hls", "mp4"),
```
Comment (dstftw) [inline youtube_dl/extractor/eagleplatform.py]: How does `secure_m3u8` look like for you? For me it never contains `m3u8` or `hls`. Can you provide any example such URL?

Comment (remitamine) [inline youtube_dl/extractor/eagleplatform.py]: it what they use in the js code.
there also a possibility for videos to have a f4m manifest they have some code for this case also.

**Comment by dstftw** [comment]
Cherry-picked, thanks.

**Pull request closed without merging**